mautic/mautic
# Add threads support for broadcast command

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR added support for threads in `mautic:broadcast:send` command.
This allow process more emails or sms in shorten time.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create sms and email for segments and set publish up date.
3. Try send it by 3 commands

`php bin/console mautic:broadcast:send --max-threads=3 --thread-id=1`
`php bin/console mautic:broadcast:send --max-threads=3 --thread-id=2`
`php bin/console mautic:broadcast:send --max-threads=3 --thread-id=3`

You should see like each command send different counts and together should send all of it.
This allow run multiple command in same time without duplicate send of emails or sms.
